src/rdf.rs

```diff
@@ -466,6 +519,190 @@ impl Graph {
     pub fn add(&mut self, triple: Triple) {
         self.triples.push(triple);
     }
+
+    pub fn as_ref(&self) -> GraphRef {
+        GraphRef {
+            triples: self.triples.iter().collect(),
+        }
+    }
+}
+
+impl<'a> GraphRef<'a> {
+    /// Remove triple matching given terms
+    pub fn take(
+        &mut self,
+        subject_opt: Option<&Subject>,
+        predicate_opt: Option<&Predicate>,
+        object_opt: Option<&Object>,
+    ) -> Option<&'a Triple> {
+        let mut i = 0;
+        while i < self.triples.len() {
+            if self.triples[i].matches(subject_opt, predicate_opt, object_opt) {
+                return Some(self.triples.swap_remove(i));
```

Comment:
I don't think so. When I try to create an example in JSON-LD of what would be two of the same statement, it results in one:
```
echo '{"a:b": ["a:c", "a:c"]}' | didkit to-rdf-urdna2015
_:c14n0 <a:b> "a:c" .
```
I think it's implied in the [RDF data model](https://www.w3.org/TR/rdf11-concepts/#data-model) being based on sets, in which abstractly an item may be in the set but not in it multiple times.
> The core structure of the abstract syntax is a set of triples, each consisting of a subject, a predicate and an object. A set of such triples is called an RDF graph.
> [...]
> An RDF triple encodes a statement—a simple logical expression, or claim about the world. An RDF graph is the conjunction (logical AND) of its triples

In this implementation though, the `triples` struct field is public, so it could insert duplicates. This function only takes the first one. The way this function is used, if there is a duplicate, that would result in a "leftover" triple which would cause an error later in the code: https://github.com/spruceid/ssi/pull/241/files#diff-859afcdde128730c6eb7badc55e61454f63fed6cbb70d7a513f70502420d57cfR487